pandas/core/indexes/extension.py

```diff
@@ -259,27 +240,6 @@ def _concat_same_dtype(self, to_concat, name):
         arr = type(self._data)._concat_same_type(to_concat)
         return type(self)._simple_new(arr, name=name)
 
-    @doc(Index.take)
-    def take(self, indices, axis=0, allow_fill=True, fill_value=None, **kwargs):
-        nv.validate_take(tuple(), kwargs)
-        indices = ensure_platform_int(indices)
-
-        taken = self._assert_take_fillable(
-            self._data,
-            indices,
-            allow_fill=allow_fill,
-            fill_value=fill_value,
-            na_value=self._na_value,
-        )
-        return type(self)(taken, name=self.name)
-
-    def unique(self, level=None):
-        if level is not None:
-            self._validate_index_level(level)
-
-        result = self._data.unique()
-        return self._shallow_copy(result)
```

Comment:
The base class ends up dispatching to `IndexOpsMixin` unique:

https://github.com/pandas-dev/pandas/blob/9c08fe1e192ed4ab197b042d8477f583d7a1cdeb/pandas/core/base.py#L1257-L1270

The `hasattr(values, "unique")` could probably be made more explicit / cleaner to check for EA, but basically this should also be the same

Reply:
> could probably be made more explicit / cleaner

sounds worthwhile, yah